### DISCUSSION TITLE
display remains dark

### DISCUSSION THREAD

**Discussion by MHz000** [rzeldent/esp32-smartdisplay#171] [General]
admittedly I am not very familiar with git and VScode so I need help at that level. I have recursively cloned https://github.com/rzeldent/esp32-smartdisplay-demo and thus received a folder esp32-smartdisplay with this name and this content:

.git .github .pio .vscod assets boards include src test -gitignore .gitmodules library.json platformio.ini README.md

The boards folder contains all booard definitions
The scr folder contains all xxx.c files from esp_io_expander.c to lvg_touch_xpt2046_spi.c
The platformio.ini folder contains
[platformio]
#default_envs = esp32-1732S019C
#default_envs = esp32-1732S019N
etc
[env]
platform = espressif32
framework = arduino

build_flags =
-Ofast
-Wall
'-D CORE_DEBUG_LEVEL=ARDUHAL_LOG_LEVEL_VERBOSE'
# LVGL settings
'-D LV_CONF_PATH=${platformio.test_dir}/lv_conf.h'
'-D ESP_LCD_PANEL_IO_ADDITIONS_VER_MAJOR=1'
'-D ESP_LCD_PANEL_IO_ADDITIONS_VER_MINOR=0'
'-D ESP_LCD_PANEL_IO_ADDITIONS_VER_PATCH=1'

lib_deps =
lvgl/lvgl@^8.3.9
# The platformio.test_dir contains the test_main.cpp just to have an setup() and loop() function
# so it will compile
${platformio.test_dir}
rzeldent/esp32_smartdisplay

lib_deps = https://github.com/rzeldent/esp32-smartdisplay.git

;---------------------------------------------------
;[env:esp32-2432S028R]
;board = esp32-2432S028R

;[env:esp32-2432S028Rv2]
;board = esp32-2432S028Rv2

[env:esp32-2432S028Rv3]
board = esp32-2432S028Rv3

I use the ESP32-2432S028 board with USB-C ubd micro
When compiling esp_lcd_touch.c no error occurs and it can be uploaded **but:**
the display remains dark and only this appears in the terminal window

--- Terminal on COM4 | 9600 8-N-1
--- Available filters and text transformations: colorize, debug, default, direct, esp32_exception_decoder, hexlify, log2file, nocontrol, printable, send_on_enter, time
--- More details at https://bit.ly/pio-monitor-filters
--- Quit: Ctrl+C | Menu: Ctrl+T | Help: Ctrl+T followed by Ctrl+H
h␑␓0��L���
h!�N��Ճ�␑��L�␀�
I'm at a loss at this point

**Comment by rzeldent** [comment]
This is just to test if it compiles. It is a library and needs to be embedded in another application.

If you clone (yes, also recursively) https://github.com/rzeldent/esp32-smartdisplay-demo it should work!

**Comment by MHz000** [comment]
**Bingo!**
Rene cloned the repo and now I see the first image on the display :-))
Thank you for your tip. This is the perfect starting point for my own development.
I have to admit that I am still confused. But now on a much higher level! 
Why couldn't I integrate esp32_smartdisplay and lvgl sunton cyd lcd tft touch in the usual way but had to clone them?
remember I am not very familiar with git and VScode 
Thank you

**Reply by rzeldent** [reply]
This has been done to make it modular and easy to integrate into your projects. Recursive cloning also includes the repository with the boards so works on a lot of different boards.
